### PR TITLE
Upgrade font processing

### PR DESCRIPTION
**Pull request by Tzenchor**
Corrects problems detected in pull #4472. The formatting changes were detected just before writing this. I did a few tweaks to my style configuration and compared what it changed from the base directly with Visual Studio, and it highlighted a lot less lines. The changes are mainly replacing tabs with spaces, sorry about that.
This pull changes the way fonts are imported. Previously the size of the font was taken from the font autohinter, and this side has proved being wrong in some cases, see #4468 with Trebuchet MS.
Now the font gets read in its entirety to measure it.
Deleted old code from days of the GDI rasterizer that was never called in method `CreateFontFace`
Removed try... catch from method `CreateFontFace` because it catched everything to just rethrow it

### CODE REVIEW

File: MonoGame.Framework.Content.Pipeline/Graphics/Font/SharpFontImporter.cs [hunk 1]
```diff
@@ -9,117 +9,137 @@
...
+                        glyphList.Add(RetrieveGlyphData(glyphIndexes[found].Key, face));
+                        //glyphIndexes.RemoveAt(found);
+                    }
+                }
+                //For each character not included in the font load the glyph with index 0
+                face.LoadGlyph(0, LoadFlags.Default, LoadTarget.Normal);
+                face.Glyph.RenderGlyph(RenderMode.Normal);
+                foreach (var character in notFoundGlyphs)
+                {
+                    //var glyph = ImportGlyph(character.Key, face);
```
Comment (dellis1972) [inline MonoGame.Framework.Content.Pipeline/Graphics/Font/SharpFontImporter.cs]: can we remove this commented line? same for the glyphIndexes on line 65 :)

Comment (Tzenchor) [inline MonoGame.Framework.Content.Pipeline/Graphics/Font/SharpFontImporter.cs]: Ups, forgot them. I'll review i have not left any more code commented

**Comment by Tzenchor** [comment]
I have just run the tests in my computer and all have passed without problems.

**Comment by tomspilman** [comment]
@Tzenchor - This PR seems to be conflicted now... can you rebase this branch?

**Comment by Tzenchor** [comment]
I think I have managed to do the rebase, I will push the changes after running some tests to check this still works

**Comment by Tzenchor** [comment]
Ok, i have just pushed and now it shows changes from the other merges since i created the branch if this is wrong just tell and I'll recreate the branch again.

If someone wonders how I made this mess, i pulled the changes from mono/develop to Tzenchor/develop, then I changed branch to Tzenchor/Upgrade-font-processing, and did a rebase followed by a pull to fix conflicts as the console told me.

**Comment by dellis1972** [comment]
For future reference I do the following when getting changes from the mg repo I to my fork

git fetch upstream
git rebase upstream/develop

Where upstream is the mg repo. I never directly pull from the mg repo it kinda messes things up.

-----Original Message-----

**Comment by tomspilman** [comment]
@dellis1972 - Is this still safe to merge like this?  Will it work itself out after the merge?

**Comment by KonajuGames** [comment]
I don't think it would be safe to merge this as it is.  The changes should be presented in a new PR and close this one.

**Comment by Tzenchor** [comment]
Ok, will do a new pull request.

**Comment by Tzenchor** [comment]
I was preparing the new pull request when I thought that I'll be making further changes to `SharpFontImporter.cs` in pull #4537, so maybe I should integrate these changes there directly.
But if you want to have this merged for version 3.5 tell me and I make the pull request

**Comment by tomspilman** [comment]
> so maybe I should integrate these changes there directly

Ideally no...   we like smaller changes and not bigger ones.  If this is independent of the other it is better to keep it that way.

**Comment by Tzenchor** [comment]
Ok, new pull incoming. As #4545
Closing

**Pull request closed without merging**